### PR TITLE
prevent decode errors when displaying failures

### LINKED ISSUE #26256: Template with lookup plugin gives wrong failure message if binary files are involved
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- template with lookup plugin to fetch files

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
 
```

##### CONFIGURATION

default settings
##### OS / ENVIRONMENT

CentOS Linux release 7.2.1511 (Core)
##### SUMMARY

Templating fails with wrong message given below if there is a binary file. This occur when template module is used with lookup plugin.
The file that caused the failure is not indicated

##### STEPS TO REPRODUCE

Include a binary file in list of files to be templated., which is fetched using lookup plugin.

```
- name: Templates all .j2 files recursively
  template: src={{ item }} dest={{base_package_dir}}/{{release_no}}/{{build_no}}/{{ item|replace(autodeploy_path , 'config') }}  
  with_items: "{{ lookup('pipe', 'find '+ autodeploy_path + '  -type f -regextype posix-egrep -regex \".*\\.(j2)$\"').split('\n') }}"
  tags: property_template_2

```

##### EXPECTED RESULTS

Expected that Ansible will indicate the file that caused the template failure. e.g. Error templating file binaryxxxx.file
##### ACTUAL RESULTS

The task fails, other files are properly templated but the files that have problems are not indicated.

```
[WARNING]: Failure using method (v2_runner_item_on_failed) in callback plugin (<ansible.plugins.callback.default.CallbackModule object at
0x359afd0>): 'ascii' codec can't decode byte 0xef in position 101: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by jctanner**
##### SUMMARY
Fixes #26256

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
callbacks

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/plugins/callback/__init__.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ def _dump_results(self, result, indent=None, sort_keys=True, keep_invocation=Fal
         if 'exception' in abridged_result:
             del abridged_result['exception']
 
+        # ensure we don't hit decode errors on displaying errors
+        abridged_result = to_text(abridged_result)
+
```
Comment (bcoca) [inline lib/ansible/plugins/callback/__init__.py]: wouldn't it be safer to put his around `result` in line 84?

Comment (jctanner) [reply lib/ansible/plugins/callback/__init__.py]: If result is sent to to_text() before strip_internal_keys() is called, then the call to strip_internal_keys() results in ...

```
 [WARNING]: Failure using method (v2_runner_item_on_failed) in callback plugin
(<ansible.plugins.callback.default.CallbackModule object at 0x7f48f2306fd0>): 'unicode' object has no
attribute 'copy'
```

Comment (bcoca) [reply lib/ansible/plugins/callback/__init__.py]: sorry i should have been more explicit, we dont really want to modify result itself if possible
```
   abridged_result = to_text(strip_internal_keys(result))
```

**Comment by jctanner** [comment]
deprecated by https://github.com/ansible/ansible/issues/26295

**Pull request closed without merging**